### PR TITLE
Make a static build of the viewer

### PR DESCRIPTION
**Pull request by Karang**

### CODE REVIEW

**Comment by rom1504** [comment]
Not sure if all will go into this PR, but we discussed about these steps:
* [ ] extract rendering API with methods like move() entitySpawn() chunkColumnLoad()
* [ ] extract the socket io client that calls the rendering API
* [ ] use https://github.com/streamich/memfs-webpack to use https://github.com/PrismarineJS/prismarine-provider-anvil
* [ ] have a github package example of pviewer (self contained frontend)

A bit later:
* [ ] make everything use the refactored rendering API
* [ ] document the rendering API
* [ ] document the socket io API
* [ ] expose the rendering API to node too (headless rendering)

Even later:
* standardize the way to provide data to the rendering API from a prismarine-world (using pworld events), allowing to have a single unified way (pworld) to provide data to the rendering API

**Comment by rom1504** [comment]
how I think the code could be structured in components:

there should be 5 independent folders
1. rendering API: contains the public/ folder and is the only one that knows about it
2. rendering API client : depends on 1
3. rendering API server: talk with 2 with a specified socket io API
4. rendering providers: depends on 3 (eg mineflayer)
5. examples : uses 1, 2, 3, or 4: sometimes not all of them

**Review by extremeheat — COMMENTED** [review]
Looks good so far, I'll try doing an electron demo and see if I run into any major issues. Then I'll look at mceditor.

Some comments from what I've seen so far:
* src/version.js depends on dot versions to determine which version to load, this can be an issue for snapshots
* bundling assets for every version will lead to large files and the size will continue to increase as more mc versions come in, even more if bedrock support is added

File: package.json [hunk 1]
```diff
@@ -7,35 +7,41 @@
     "test": "jest --verbose --runInBand --forceExit",
     "pretest": "npm run lint && require-self",
     "lint": "standard",
-    "prepare": "npm install require-self && require-self",
+    "prepare": "npm install require-self && require-self && webpack",
```
[the remark this answers is not in the document]
Comment (rom1504) [reply package.json:10]: Webpack would run only for maintainers and contribution of prismarine-viewer here, not for users and dependent packages.
Webpack produces some file which are then bundled in npm package that people can use with whatever tech they like

**Comment by rom1504** [comment]
To support all versions we need to bundle all versions.
The current size of the compressed file is 5MB, that seems ok.
We could provide options to bundle a single file if there is an use case, but for now it seems the most simple option.

**Comment by Karang** [comment]
I added an example for pviewer standalone in browser, it uses a simple generator. Using anvil would need some refactoring in provider-anvil first so it is outside the scope of this PR.

So i think what is left is finishing @rom1504 PR to rename files

**Comment by rom1504** [comment]
Great, I'll finish my PR soon ( will rebase on your latest changes)

**Review by rom1504 — APPROVED** [review]

**Pull request merged**